### DISCUSSION TITLE
how to Link two computers with vba-m?

### DISCUSSION THREAD

**Discussion by ghost** [visualboyadvance-m/visualboyadvance-m#1162] [General]
how to Link two computers with vba-m?

**Comment by rkitover** [comment]
Download the nightly on both computers from http://win.vba-m.com/nightly/.

Open the GBA game with link support you want to play on both computers.

On both computers:

Go to `Options -> Link -> Type` and set `Cable`.

On computer 1:

Go to `Options -> Link -> Start Network Link`, choose `Server` and press `Connect`.

On computer 2:

Go to `Options -> Link -> Start Network Link`, choose `Client`

In the `Server` field, enter the IP address of the other computer.

Press `Connect`.

**Comment by ghost** [comment]
we done this but it has an error

**Comment by rkitover** [comment]
What is the error? What game are you trying to play? Is it for GB or GBA?

**Comment by XKirby** [comment]
I can attest to this. With every version of the Oct 19th 11PM nightly build, regardless of game played (and even if no game is loaded), a friend and I ran into an error on the Client trying to connect to the Host. It only shows a tiny window that says "Error occured. Please try again." The games we tried were Mega Man Battle Network 6 and Mega Man Battle Network 3, but I'm assuming there's other underlying issues.

On a single PC it works just fine on Link Cable mode for Battle Network 6 (I didn't try it myself with BN3). Even with recommended settings suggested at this github, I haven't been able to connect to any other computers. On the host's side, it says "All Players Connected" on the debug builds and acts as if I had connected, while on the client's side it throws that error that was mentioned. Unfortunately the half-connection is super unstable and will most likely softlock or crash the emulator upon anything to do with multiplayer menus.

When it connects successfully on a single computer with the debug builds, it throws a lot of warnings in the console window about partial packets. This seems to slow down the game, but besides that the connection stays until I mess with the window. When it fails and shows the error pop-up, nothing is displayed in the console. Not even an attributing reason for the error.

Hopefully this giant block of text can be somewhat helpful. I just spent a few hours trying as many off-shot variables I could to make it work, with no luck.

**Comment by rkitover** [comment]
@XKirby were you trying on a LAN? If not, did you forward the port?

**Reply by MrEnd0less** [reply]
3 Years later (or 4), I'm having the same issue as @ghost and @XKirby, even when following the guide on @rkitover 's first reply.
I am attempting to play Pokemon Quetzal (A.k.a. Pokemon Emerald Multiplayer) with a friend online. We are on two different computers, in different networks.

We tried two methods, both of which gave errors:

1. Using hamachi to simulate being in the same network, which allowed me to start the server, but didn't allow my friend to connect to it, giving the error message "Error occurred. Please try again."

Then I decided to
2. Port forward, which after I managed to set up properly, didn't allow my friend to connect either (I checked with various tools to make sure the port really is open). The emulator *did* let me start the server like before, but my friend couldn't join. I gave my friend my router's external IP of course, with the correct port, but hosted at my computer's private IP, and still my friend got the same error message "Error occurred. Please try again."

Additional information:
I use a MacBook Pro, macOS Mojave (10.14.6).
My friend uses an Asus gaming laptop, Windows 10.
We are both using VBA-M 2.1.8.
We are in different countries, so local play is not an option unfortunately.
We are trying to avoid having to use Parsec to play, but it's the only option we have left so far.

**Reply by ZachBacon** [reply]
Network connections over the internet are not recommended at this time.

**Reply by ttkblock** [reply]
> 3 Years later (or 4), I'm having the same issue as @ghost and @XKirby, even when following the guide on @rkitover 's first reply. I am attempting to play Pokemon Quetzal (A.k.a. Pokemon Emerald Multiplayer) with a friend online. We are on two different computers, in different networks.
> 
> We tried two methods, both of which gave errors:
> 
> 1. Using hamachi to simulate being in the same network, which allowed me to start the server, but didn't allow my friend to connect to it, giving the error message "Error occurred. Please try again."
> 
> Then I decided to 2. Port forward, which after I managed to set up properly, didn't allow my friend to connect either (I checked with various tools to make sure the port really is open). The emulator _did_ let me start the server like before, but my friend couldn't join. I gave my friend my router's external IP of course, with the correct port, but hosted at my computer's private IP, and still my friend got the same error message "Error occurred. Please try again."
> 
> Additional information: I use a MacBook Pro, macOS Mojave (10.14.6). My friend uses an Asus gaming laptop, Windows 10. We are both using VBA-M 2.1.8. We are in different countries, so local play is not an option unfortunately. We are trying to avoid having to use Parsec to play, but it's the only option we have left so far.

@MrEnd0less OMG I finally found this discussion. Me and my friend having the exact same situation that you describe above. We try to link Pokemon Quetzal, we're on different countries. The only different is we successfully connect with each other using Port Forward method. But unfortunately when we try to link and start a multiplay option in Pokemon Quetzal, I stuck on ''press A to connect, B to cancel'' screen. And when I press the ''A'' button, my friend's screen just suddenly crashed.

Our Link timeout setting is ''999999'' (in millliseconds), Server port 5738. We're both on Mac, I'm on masOS Monterey(12.4) and we're both on VBA-M version (2.1.11). Still trying to figure out how to fix that, if you have any new update on your situation please share with me, I'm just very glad that me and my friend are not the only one having this issue.

**Comment by XKirby** [comment]
We tried both simulated LAN via Radmin VPN and port-forwarding. Nothing worked.

**Comment by ghost** [comment]
https://github.com/visualboyadvance-m/visualboyadvance-m/issues/786

**Comment by DarkPrincess638** [comment]
What port is it supposed to work on? The port number is pretty important and I can't find a single listing of it anywhere.

**Comment by Renpona** [comment]
> What port is it supposed to work on? The port number is pretty important and I can't find a single listing of it anywhere.

It seems to default to port 5738. 

It'll connect to a server on that port, but it waits for a specific response from the server before VBA-M will consider the connection complete, and I haven't figured out exactly what the server needs to send back yet.

**Comment by johnnieb333** [comment]
the Start Network Link is grayed out on every release , and the nightly that you want to link us to is no longer there , I am looking for a emulator that will allow for trading on 1 pc , but the option has been taken away , please explain

**Comment by Squall-Leonhart** [comment]
You can't start a network link if you haven't selected a connection type.

**Comment by Reshirou** [comment]
Bump

**Comment by loobage** [comment]
does anyone know wy i got this when i press the confirm button 
![image](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/149265277/6c07283c-f111-4ac1-a371-47dd2478d9e4)
